File: app/src/main/java/com/organizer/layouts/add/AddLayout.java
package com.organizer.layouts.add;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.GradientDrawable;
import android.view.View;
import androidx.annotation.NonNull;
import com.organizer.layouts.BaseLayout;

public class AddLayout extends BaseLayout
{
    public AddLayout(Context context)
    {
        super(context);
        
        GradientDrawable drawable = new GradientDrawable();
        drawable.setTint(Color.BLACK);
        drawable.setAlpha(125);
        setBackground(drawable);
        
        AddTaskLayout addTaskLayout = new AddTaskLayout(context);
        addView(addTaskLayout);
        
        AddEventLayout addEventLayout = new AddEventLayout(context);
        addView(addEventLayout);
    }
    
    @Override
    protected void onLayout(boolean changed, int left, int top, int right, int bottom)
    {
        int width = right - left;
        int height = bottom - top;
        float ratio = (float) Integer.max(width, height) / Integer.min(width, height);
        int widthMargin = ((int) (width / (ratio * (height >= width ? 3 : 2))));
        int heightMargin = ((int) (height / (ratio * (width > height ? 3 : 2))));
        
        for (int i = 0; i < getChildCount(); i++)
        {
            getChildAt(i).layout(left + widthMargin / 2, top + heightMargin / 2, right - widthMargin / 2, bottom - heightMargin / 2);
        }
    }
    
    @Override
    protected void onVisibilityChanged(@NonNull View changedView, int visibility)
    {
        if (visibility == INVISIBLE)
        {
            for (int i = 0; i < getChildCount(); i++)
            {
                getChildAt(i).setVisibility(View.INVISIBLE);
            }
        }
    }
    
    public void show(Object addLayoutClass)
    {
        for (int i = 0; i < getChildCount(); i++)
        {
            if (getChildAt(i).getClass() == addLayoutClass)
            {
                getChildAt(i).setVisibility(View.VISIBLE);
            }
        }
    }
}

File: app/src/main/kotlin/com/organizer/layouts/add/AddLayout.kt
package com.organizer.layouts.add

import android.graphics.Color
import android.graphics.drawable.GradientDrawable
import android.view.View
import com.organizer.layouts.Layout
import kotlin.reflect.KClass

class AddLayout : Layout() {

    init {
        val drawable = GradientDrawable()
        drawable.setTint(Color.BLACK)
        drawable.alpha = 125
        background = drawable

        addView(AddTaskLayout())
        addView(AddEventLayout())
    }

    override fun onLayout(changed: Boolean, left: Int, top: Int, right: Int, bottom: Int) {
        val width = right - left
        val height = bottom - top
        val ratio = Integer.max(width, height).toFloat() / Integer.min(width, height)
        val widthMargin = (width / (ratio * if (height >= width) 3 else 2)).toInt()
        val heightMargin = (height / (ratio * if (width > height) 3 else 2)).toInt()
        for (i in 0 until childCount) {
            getChildAt(i).layout(
                left + widthMargin / 2,
                top + heightMargin / 2,
                right - widthMargin / 2,
                bottom - heightMargin / 2
            )
        }
    }

    override fun onVisibilityChanged(changedView: View, visibility: Int) {
        if (visibility == INVISIBLE) {
            for (i in 0 until childCount) {
                getChildAt(i).visibility = INVISIBLE
            }
        }
    }

    fun show(addLayoutClass: KClass<*>) {
        for (i in 0 until childCount) {
            if (getChildAt(i)::class == addLayoutClass) {
                getChildAt(i).visibility = VISIBLE
            }
        }
    }
}
